tests/vm_test.go

```diff
@@ -698,6 +698,34 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 				Expect(err.Error()).To(Equal("Error stopping VirtualMachine VM is not running"))
 			})
 
+			It("[test_id:3007]Should force restart a VM with terminationGracePeriodSeconds>0", func() {
+
+				By("getting a VM with high TerminationGracePeriod")
+				newVMI := tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskFedora))
+				gracePeriod := int64(1800)
+				newVMI.Spec.TerminationGracePeriodSeconds = &gracePeriod
+				newVM := tests.NewRandomVirtualMachine(newVMI, true)
+				_, err := virtClient.VirtualMachine(newVM.Namespace).Create(newVM)
+				Expect(err).ToNot(HaveOccurred())
+				waitForVMIStart(virtClient, newVMI)
+
+				oldCreationTime := newVMI.ObjectMeta.CreationTimestamp
+				oldVMIUuid := newVM.ObjectMeta.UID
+
+				By("Invoking virtctl --force restart")
+				forceRestart := tests.NewRepeatableVirtctlCommand(vm.COMMAND_RESTART, "--namespace", newVM.Namespace, "--force", newVM.Name, "--gracePeriod=0")
+				err = forceRestart()
+				Expect(err).ToNot(HaveOccurred())
+
+				waitForVMIScheduling(virtClient, newVMI)
```

Comment:
I think that this is too coarse. You  should check for the following in the tests:
 *  that you either see a Pod which still exists, but has a `DeletionGracePeriodSeconds` in the object metadata set to 0 and the deletion timestamp is not nil (so this is the case where the force delete request happened, but the pod is still not gone)
 * that the pod is already gone after the request (that can happen when kubelet reacts really fast on the removal).

If you are not also checking the pods, then you are only checking that a restart happened in general, but not if it was forced.

Reply:
Added check for both.